src/Java.Interop.BootstrapTasks/Java.Interop.BootstrapTasks/JdkInfo.cs

```diff
@@ -158,24 +213,72 @@ string GetJavaHomePathFromEnvironment ()
 			return null;
 		}
 
+		// macOS
+		string GetJavaHomePathFromLibexec ()
+		{
+			var java_home	= Path.GetFullPath ("/usr/libexec/java_home");
+			if (!File.Exists (java_home)) {
+				return null;
+			}
+			string path = null;
+			Exec (java_home, "", (o, e) => {
+					if (string.IsNullOrEmpty (e.Data))
+						return;
+					Log.LogMessage (MessageImportance.Low, $"    {e.Data}");
+					path = e.Data;
+			});
+			return path;
+		}
+
+		// Linux
+		string GetJavaHomePathFromAlternatives ()
+		{
+			var alternatives  = Path.GetFullPath ("/etc/alternatives/java");
+			if (!File.Exists (alternatives))
+				return null;
+			string path = null;
+			Exec ("readlink", $"\"{alternatives}\"", (o, e) => {
+					if (string.IsNullOrEmpty (e.Data))
+						return;
+					Log.LogMessage (MessageImportance.Low, $"    {e.Data}");
+					path = e.Data;
+			});
+			return path;
```

Comment:
`/etc/alternatives/java` points to the java **executable**, not home path:

```shell
$ readlink /etc/alternatives/java
/usr/lib/jvm/java-8-oracle/jre/bin/java
```

so more processing is necessary here to match output from e.g. macOS `java_home`